api/internals/init.md

```diff
@@ -0,0 +1,169 @@
+3. The repo root is writeable.
+4. The repo either doesn't exist and hasn't been initialized, or does already exist but the `--force` flag is supplied.
+
+If all of these are satisfied, output `"initializing ipfs node at %s\n", repoRoot` to standard out and proceed.
+
+### 2. Generate Repo Configuration
+
+These are the steps to producing the default [config tree](https://github.com/ipfs/specs/tree/master/repo#config). The actual output format will depend on the repo backend (i.e. likely JSON if fs-repo). Check the specification for the relevant backend if unsure.
+
+The following sections describe the various subtrees in the config and how to generate them. For simplicity and consistency, JSON will be used.
+
+#### Identity
+
+The RSA keypair must be >= 1024 bits.
+
+Before beginning key generation, write `"generating %v-bit RSA keypair...", nbits` to standard out. (no newline)
+
+Generate the keypair. If successful, append `"done\n"` to standard out.
+
+The private key must be formatted as the base64 encoding of its bytes. It is currently stored unencrypted.
+
+The PeerID must be formatted as
+
+```json
+Base58.Encode(Multihash('SHA2-256', PublicKey.Bytes))
+```
+
+Resulting in the following config tree entry:
+
+```json
+"Identity": {
+	"PeerID": "<PeerID>",
+	"PrivKey": "<PrivKey>"
+}
+```
+
+Finally, write `"peer identity: %s\n", ident.PeerID` to standard out.
+
+#### Bootstrap Peers
+
+The default bootstrap peers form the following config tree:
+
+```json
+"Bootstrap": [
+  "/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+  "/ip4/104.236.176.52/tcp/4001/ipfs/QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z",
+  "/ip4/104.236.179.241/tcp/4001/ipfs/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
+  "/ip4/162.243.248.213/tcp/4001/ipfs/QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm",
+  "/ip4/128.199.219.111/tcp/4001/ipfs/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu",
+  "/ip4/104.236.76.40/tcp/4001/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64",
+  "/ip4/178.62.158.247/tcp/4001/ipfs/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
+  "/ip4/178.62.61.185/tcp/4001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3",
+  "/ip4/104.236.151.122/tcp/4001/ipfs/QmSoLju6m7xTh3DuokvT3886QRYqxAzb1kShaanJgW36yx"
+]
+```
+
+#### Version
+
+This is the `"Version"` config subtree, which is just the key `"Current"`, which
+evaluates to the IPFS program's current version literal (e.g. `"0.4.0-dev"`.
+
+```json
+"Version": {
+  "Current": "v0.4.0-dev"
+}
+```
+
+#### Miscellaneous
+
+There are some other various static values that also must be included in the config tree:
+
+```json
+// setup the node's default addresses.
+// Note: two swarm listen addrs, one tcp, one utp.
+"Addresses": {
+	"Swarm": [
+		"/ip4/0.0.0.0/tcp/4001",
+		"/ip6/::/tcp/4001"
+	],
+	"API": "/ip4/127.0.0.1/tcp/5001",
+	"Gateway": "/ip4/127.0.0.1/tcp/8080"
+},
```

Comment:
this is specific to go-ipfs. does it belong in this spec? (maybe, maybe not, not sure)